octree/include/pcl/octree/octree_base.h

```diff
@@ -162,6 +165,24 @@ namespace pcl
           return BreadthFirstIterator (this, 0, NULL);
         };
 
+        // Octree breadth iterators at a given depth
+        typedef OctreeFixedDepthIterator<OctreeT> FixedDepthIterator;
+        typedef const OctreeFixedDepthIterator<OctreeT> ConstFixedDepthIterator;
+
+        FixedDepthIterator fixed_depth_begin (unsigned int fixed_depth_arg = 0u)
+        {
+          if (this->octree_depth_ < fixed_depth_arg)
+          {
+            PCL_WARN ("[pcl::octree::FixedDepthIterator] The requested fixed depth was bigger than the octree's depth.");
+            PCL_WARN ("[pcl::octree::FixedDepthIterator] fixed_depth = %d (instead of %d)", this->octree_depth_, fixed_depth_arg);
+          }
+          return FixedDepthIterator (this, std::min (fixed_depth_arg, this->octree_depth_));
```

Comment:
This check is also performed inside `reset()` function. I think we should get rid of it here and move warnings there. Because when user manually creates an iterator with wrong depth, a warning is also helpful.